### PR TITLE
P44: only hash_map macro

### PR DESCRIPTION
**Pull request by alvarogarcia7**

### CODE REVIEW

File: p44/src/hashmap_macro.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// Write an integration test fo the macro
+
+#[macro_export]
+// Source: src/macros.rs - vec!
+macro_rules! hashmap {
+    () => {
+        HashMap::<u64, bool>::new();
+    };
+    ($($key:expr => $value:expr $(,)?)+) => {
+        HashMap::<u64, bool>::from([$(($key, $value) ,)*]);
```
Comment (newpavlov) [inline p44/src/hashmap_macro.rs]: This macro should work with any types of keys and values. Plus, you unnecessarily create an array here. Try to write code using the `HashMap::insert` method.

File: p44/NOTES.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+2024-06-07 12:57:04 AGB
+
+# Generic in macros
+
+Possible enhancement:
+
+```rust
+hashmap::<u64, u64> ! (
```
Comment (newpavlov) [inline p44/NOTES.md:8]: Such syntax is not possible with declarative macros.

File: p44/src/hashmap_macro.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    };
+}
+
+#[macro_export]
+// Source: src/macros.rs - vec!
+macro_rules! hashmap_poorgeneric {
+    ( $(<)+ $key_type:ty $(,)+ $value_type:ty $(>)+)  => {
```
Comment (newpavlov) [inline p44/src/hashmap_macro.rs]: You do not need to provide type annotation to this macro. If needed, user can provide type annotations themselves, e.g. `let x: HashMap<u64, u32> = hashmap!( 1=> 3, 2 => 4)`.

File: p44/src/hashmap_macro.rs (hunk 1 above)
Comment (alvarogarcia7) [reply p44/src/hashmap_macro.rs]: 1. types: ok - done
2. array - done

File: p44/NOTES.md (hunk 2 above)
Comment (alvarogarcia7) [reply p44/NOTES.md:8]: ok - any alternative?

Comment (newpavlov) [reply p44/NOTES.md:8]: You alredy implemented it. :)

Comment (alvarogarcia7) [reply p44/NOTES.md:8]: ok - I understood - delegate the type to the assigned variable. it'll be inferred.

**Review by newpavlov — APPROVED** [review]

File: p44/src/hashmap_macro.rs (hunk 3 above)
Comment (alvarogarcia7) [reply p44/src/hashmap_macro.rs]: done

**Pull request merged**